File: 백준 1300번 K번째수.cpp
#include<iostream>
#include<algorithm>
using namespace std;
long long n,k,answer;
int main()
{
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cin>>n>>k;
	long long left(1),right(n*n);
	while(left<=right)
	{
		long long mid((left+right)/2);
		long long cnt(0);
		for(long long i=1; i<=n; i++)
		cnt+=min(n,mid/i);
		if(cnt>=k)
		{
			answer=mid;
			right=mid-1;
		}else
		left=mid+1;
	}
	cout<<answer;
}

File: Boj-Kotlin&Java/백준 1300번 K번째 수.kt
package 백준

import kotlin.math.min
private var answer:Long=0
fun main() {
    val br=System.`in`.bufferedReader()
    val n=br.readLine().toInt()
    val k=br.readLine().toInt()
    var left:Long=1
    var right:Long=k.toLong()
    var cnt=0
    while (left<=right){
        cnt=0
        val mid=(left+right)/2
        for(i in 1..n)
            cnt+= min((mid/i),n.toLong()).toInt()
        if(k<=cnt){
            answer=mid
            right=mid-1
        }else
            left=mid+1
    }
    println(answer)
}